src/wallet/test/availablecoins_tests.cpp

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2022 The Bitcoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or https://www.opensource.org/licenses/mit-license.php.
+
+#include <validation.h>
+#include <wallet/coincontrol.h>
+#include <wallet/spend.h>
+#include <wallet/test/util.h>
+#include <wallet/test/wallet_test_fixture.h>
+
+#include <boost/test/unit_test.hpp>
+
+namespace wallet {
+BOOST_FIXTURE_TEST_SUITE(availablecoins_tests, WalletTestingSetup)
+class AvailableCoinsTestingSetup : public TestChain100Setup
+{
+public:
+    AvailableCoinsTestingSetup()
+    {
+        CreateAndProcessBlock({}, {});
+        wallet = CreateSyncedWallet(*m_node.chain, m_node.chainman->ActiveChain(), m_args, coinbaseKey);
+    }
+
+    ~AvailableCoinsTestingSetup()
+    {
+        wallet.reset();
+    }
+    CWalletTx& AddTx(CRecipient recipient)
+    {
+        CTransactionRef tx;
+        bilingual_str error;
+        CCoinControl dummy;
+        FeeCalculation fee_calc_out;
+        {
+            constexpr int RANDOM_CHANGE_POSITION = -1;
+            std::optional<CreatedTransactionResult> txr = CreateTransaction(*wallet, {recipient}, RANDOM_CHANGE_POSITION, error, dummy, fee_calc_out);
```

Comment:
No change needed: Don't quite understand why the `RANDOM_CHANGE_POSITION` is defined specifically like this. Wouldn't this just work?

```suggestion
            std::optional<CreatedTransactionResult> txr = CreateTransaction(*wallet, {recipient}, /*change_pos=*/-1, error, dummy, fee_calc_out);
```

Reply:
iirc, i copied this from other tests which use `CreateTransaction`. personally, i prefer it to just `-1` because it's more clear to me having a `NAME_CONSTANT`, but i don't have a super strong feelings about it